questionGenerator.py

```diff
@@ -0,0 +1,318 @@
+    def __init__(self, config, videoData, topicModeller, load=True):
+        self.config = config
+
+        self.clusteredTopics = getClusteredTopics(videoData, topicModeller)
+        self.dominantTopics = getDominantTopic(self.clusteredTopics)
+        self.relevantText, self.questionQueryText = getTextAndQuestion(
+            self.dominantTopics, videoData
+        )
+        self.responseData = None
+
+        if load:
+            self.loadQuestionData()
+        else:
+            self.OpenAIChatBot = OpenAIBot(self.config)
+            self.getQuestionData()
+            config.questionTokenCount = self.OpenAIChatBot.tokenUsage
+            printAndLog(
+                f"Question Generation Token Count: {config.questionTokenCount}",
+            )
+            self.saveQuestionData()
+
+    def loadQuestionData(self):
+        """
+        Loads the question data from a file.
+        """
+        self.responseData = dataLoader(
+            self.config, "questionData", self.config.videoToUse
+        )
+
+    def saveQuestionData(self):
+        """
+        Saves the question data to a file.
+        """
+        dataSaver(
+            self.responseData,
+            self.config,
+            "questionData",
+            self.config.videoToUse,
+        )
+
+    def getQuestionData(self):
+        """
+        Generates the question data using the OpenAI chatbot.
+
+        This is because this is a much simpler task than the Topic Extraction part.
+        Using LangChain in BERTopic lets BERTopic handle a lot of the complexities in how the topics are generated by leveraging LangChain.
+        THat process involves passing keywords, relevant documents, and such to get the human-interpretable topics.
+
+        In the case of Question generation, we only need to send one query per question we need to generate.
+        We can get much finer control on the prompt directly because of the simpler query call made.
+
+        LangChain is just a complicated way of building the prompt and storing information at its fundamental level as well, it just hides it from us through abstraction.
+        This is a bot of an oversimplification, but it's a good way to think about it.
+        So in this case, it made sense to stick to an OpenAIBot for our call rather than using LangChain.
+
+        https://www.singlestore.com/blog/beginners-guide-to-langchain/
+        """
+        self.responseData = {}
+        for topic in self.questionQueryText:
+            response = self.OpenAIChatBot.getResponse(self.questionQueryText[topic])
+            self.responseData[topic] = response[0]
+
+    def printQuestions(self):
+        """
+        Prints the question data.
+        """
+        for topic in self.responseData:
+            response = self.responseData[topic]
+            response = response.strip("` \n")
+            if response.startswith("json"):
+                response = response[4:]
+            parsedResponse = json.loads(response)
```

Comment:
This step here, underneath `loads` implementation raises error like for example `JSONDecodeError`. Wheneven you are using a pkg object functions need to account for how are errors are handled. Unhappy Path consideration.